config/main.py

```diff
@@ -2064,6 +2064,538 @@ def delete_snmptrap_server(ctx, ver):
+                clicommon.run_command("systemctl restart snmp.service", display_cmd=False)
+            except SystemExit as e:
+                click.echo("Restart service snmp failed with error {}".format(e))
+                raise click.Abort()
+        else:
+            click.echo("SNMP Location {} does not exist.  The location is {}".format(location, snmp['LOCATION']['Location']))
+            sys.exit(1)
+    except KeyError:
+        if "LOCATION" not in snmp.keys():
+            click.echo("SNMP Location {} is not configured".format(location))
+            sys.exit(2)
+
+
+@location.command('modify')
+@click.argument('location', metavar='<location>', required=True, nargs=-1)
+@clicommon.pass_db
+def modify_location(db, location):
+    """ Modify snmp location"""
+    if isinstance(location, tuple):
+        location = " ".join(location)
+    elif isinstance(location, list):
+        location = " ".join(location)
+    snmp = db.cfgdb.get_table("SNMP")
+    try:
+        snmp_location = snmp['LOCATION']['Location']
+        if location in snmp_location:
+            click.echo("SNMP location {} already exists".format(location))
+            sys.exit(1)
+        else:
+            db.cfgdb.mod_entry('SNMP', 'LOCATION', {'Location': location})
+            click.echo("SNMP location {} modified in configuration".format(location))
+            try:
+                click.echo("Restarting SNMP service...")
+                clicommon.run_command("systemctl reset-failed snmp.service", display_cmd=False)
+                clicommon.run_command("systemctl restart snmp.service", display_cmd=False)
+            except SystemExit as e:
+                click.echo("Restart service snmp failed with error {}".format(e))
+                raise click.Abort()
+    except KeyError:
+        click.echo("Cannot modify SNMP Location.  You must use 'config snmp location add command <snmp_location>'")
+        sys.exit(2)
+
+
+@snmp.group(cls=clicommon.AbbreviationGroup)
+@clicommon.pass_db
+def user(db):
+    pass
+
+
+@user.command('add')
+@click.argument('user', metavar='<snmp_user>', required=True)
+@click.argument('user_type', metavar='<noAuthNoPriv|AuthNoPriv|Priv>', required=True)
+@click.argument('user_permission_type', metavar='<RO|RW>', required=True)
+@click.argument('user_auth_type', metavar='<MD5|SHA|HMAC-SHA-2>', required=False)
+@click.argument('user_auth_password', metavar='<auth_password>', required=False)
+@click.argument('user_encrypt_type', metavar='<DES|AES>', required=False)
+@click.argument('user_encrypt_password', metavar='<encrypt_password>', required=False)
+@clicommon.pass_db
+def add_user(db, user, user_type, user_permission_type, user_auth_type, user_auth_password, user_encrypt_type, 
+             user_encrypt_password):
+    """ Add snmp user"""
+    click.echo(user)
+    if not snmp_username_check(user):
+        click.echo("Bad SNMP user")
```

Comment:
>Bad SNMP user [](http://example.com/codeflow?start=20&length=13)

This message is not friendly to user...

Reply:
I didn't add more info there because the helper function "snmp_username_check" was providing that info above that statement.  
Per what you added above it looks like you'd prefer that I put all the logic in the "add_user" function and not the helper function.   
I think it makes more sense to put in in the helper function instead because that has the more specific info.  What do you think?

Reply:
removed this click.echo.